### PR TITLE
Update changes due to refactor dataset add data flow

### PR DESCRIPTION
**Pull request by santoshamohan**
Reason for this pull request:
During sync process of datasets using dea-sync, attribute error **AttributeError: module 'datacube.scripts.dataset' has no attribute 'load_datasets'** was raised.
This error was due to changes for refactoring of dataset add data workflow in **datacube.scripts.dataset.py** file.

Proposed changes

1. Modify **add_dataset** function to be inline with changes to **datacube.scripts.dataset.py** file.
2. Improve logging in **_find_uri_mismatches** function within /**digitalearthau/sync/scan.py** file.

### CODE REVIEW

**Comment by omad** [comment]
Can you please have a look at this @Kirill888 . I think it should be possible to make this update without importing anything from `datacube.scripts` now, but don't know the exact details.

Thanks

File: digitalearthau/index.py [hunk 1]
```diff
@@ -65,21 +70,23 @@ def add_dataset(index: Index, dataset_id: uuid.UUID, uri: str):
...
-    else:
-        raise RuntimeError('Dataset not found at path: %s, %s' % (dataset_id, uri))
+    yaml_path = uri_to_local_path(uri)
+
+    ds_resolve = Doc2Dataset(index)
+    doc_stream = ui_path_doc_stream([yaml_path], logger=_LOG, uri=True)
+    dss = dataset_script.dataset_stream(doc_stream, ds_resolve)
+
+    try:
+        dataset_script.index_datasets(dss,
```
Comment (Kirill888) [inline digitalearthau/index.py]: `index_datasets` isn't doing much, it's really just `map(index.datasets.add, dss)` with some exception guards and logging. I would prefer we didn't depend on functionality inside scripts, they are not meant to have stable APIs. This is the case where code duplication is preferable to extra dependencies.

**Review by Kirill888 — DISMISSED** [review]
We should try to get rid of all imports from `datacube.scripts`

Also is this the only place where such imports happen?

Probably worthwhile changing them all in one pull request.

**Comment by Kirill888** [comment]
Actually only other import from scripts is ingest, we can leave that one as is for now.

**Comment by santoshamohan** [comment]
All references to datacube.scripts import has been removed.

**Review by omad — APPROVED** [review]

File: digitalearthau/index.py (hunk 1 above)
Comment (santoshamohan) [reply digitalearthau/index.py]: Updated the code to remove index_datasets

**Pull request merged**